### PR TITLE
L3DSubMesh: don't preserve state if texture changes

### PR DESCRIPTION
**Pull request by bwrsandman**
Fix s_diffuse already set error.

### CODE REVIEW

**Comment by bwrsandman** [comment]
This should fix @raffclar error

File: src/3D/L3DSubMesh.cpp [hunk 1]
```diff
@@ -161,23 +161,27 @@ void L3DSubMesh::Submit(uint8_t viewId, ShaderProgram& program, uint64_t state,
...
 				texture = &Game::instance()->GetMeshPack().GetTexture(prim.skinID);
 
-			program.SetTextureSampler("s_diffuse", 0, *texture);
+			if (lastTexture != texture)
+				program.SetTextureSampler("s_diffuse", 0, *texture);
 		}
 
 		_indexBuffer->Bind(prim.indicesCount, prim.indicesOffset);
-		bgfx::submit(viewId, program.GetRawHandle(), 0, preserveState || std::next(it) != _primitives.end());
+		bgfx::submit(viewId, program.GetRawHandle(), 0, lastTexture == texture && (preserveState || std::next(it) != _primitives.end()));
```
Comment (handsomematt) [inline src/3D/L3DSubMesh.cpp]: this line concerns me, the point of state preservation is to maintain the transforms and other state information, does this not screw with that?

Comment (bwrsandman) [reply src/3D/L3DSubMesh.cpp]: Texture changes qualify as state changes in bgfx it seems

**Comment by bwrsandman** [comment]
Needs fix. The code needs to look at the next texture, not the previous.

**Pull request merged**